common/test/acceptance/pages/lms/problem.py

```diff
@@ -19,6 +19,7 @@ def problem_name(self):
         """
         Return the current problem name.
         """
+        self.wait_for_element_presence('.problem-header', 'wait for problem header')
```

Comment:
This kind of statement should be in the test, not the property. Wherever you are using this property, you should have the wait statement right before it. That, or the is_brower_on_page for that page needs to be smarter.